pkg/api/alerting.go

```diff
@@ -281,6 +281,11 @@ func CreateAlertNotification(c *models.ReqContext, cmd models.CreateAlertNotific
 func UpdateAlertNotification(c *models.ReqContext, cmd models.UpdateAlertNotificationCommand) Response {
 	cmd.OrgId = c.OrgId
 
+	err := fillWithSecureSettingsData(&cmd)
+	if err != nil {
+		return Error(500, "Failed to update alert notification", err)
+	}
+
```

Comment:
```suggestion

	// remove unencrypted settings
	if url, ok := cmd.SecureSettings["url"]; ok && url != "" {
		cmd.Settings.Del("url")
	}
	if token, ok := cmd.SecureSettings["token"]; ok && token != "" {
		cmd.Settings.Del("token")
	}

```

Reply:
Should I remove unencrypted setting when update notification setting?

Reply:
Since this only applies to slack I would like to wait with this.

Reply:
No problem, this is option for me. Newly created notification doesn't hold unencrypted token, it is safe.